emcc.py

```diff
@@ -2785,9 +2785,10 @@ def get_full_import_name(name):
   if settings.WASM_EXCEPTIONS:
     settings.REQUIRED_EXPORTS += ['__trap']
 
-  # When ASSERTIONS is set, we include stack traces in Wasm exception objects
-  # using the JS API, which needs this C++ tag exported.
-  if settings.ASSERTIONS and settings.WASM_EXCEPTIONS:
+  # When ASSERTIONS or EXCEPTION_STACK_TRACES is set, we include stack traces in
+  # Wasm exception objects using the JS API, which needs this C++ tag exported.
+  if (settings.ASSERTIONS or settings.EXCEPTION_STACK_TRACES) and \
+     settings.WASM_EXCEPTIONS:
```

Comment:
Instead of doing this can we do:

```
if settings.ASSERTIONS:
  default_setting('EXCEPTION_STACK_TRACES', 1)
```

Then we can revert the other places in this PR that check both.

Reply:
Done

Reply:
But this doesn't let me to delete other places, because setting `EXCEPTION_STACK_TRACES` doesn't mean we set `ASSERTIONS`.

Reply:
Oh I think instead I can change `#if ASSERTIONS` to not `#if ASSERTIONS || EXCEPTION_STACK_TRACES` but simply `#if EXCEPTION_STACK_TRACES`. Was this what you meant?

Reply:
(I posted and deleted several comments because they were not very well thought out)

I think adding this is fine:
```python
if settings.ASSERTIONS:
  default_setting('EXCEPTION_STACK_TRACES', 1)
```
And this can change a few places in the JS library where we do
```js
#if ASSERTIONS || EXCEPTION_STACK_TRACES
```
to simply
```js
#if EXCEPTION_STACK_TRACES
```

But this doesn't let me delete this
```python
  if (settings.ASSERTIONS or settings.EXCEPTION_STACK_TRACES) and \
     settings.WASM_EXCEPTIONS:
```
because setting `EXCEPTION_STACK_TRACES` doesn't set `ASSERTIONS`. And the point of this PR is we display stack traces even when `ASSERTIONS` is not set.

Reply:
How about just: `if settings.EXCEPTION_STACK_TRACES and  settings.WASM_EXCEPTIONS:`?  

Or are the two lines below also needed when `ASSERTIONS` are enabled on their own?   (Why?)

Reply:
About this:
```python
if settings.ASSERTIONS:
  default_setting('EXCEPTION_STACK_TRACES', 1)
```
I think the second line should be
```python
settings.EXCEPTION_STACK_TRACES = 1
```
because we have to also override user settings. For example,
```console
emcc -fwasm-exceptions -sASSERTIONS -sEXCEPTION_STACK_TRACES=0 test.cpp
```
should also print stack traces. I thought about not printing stack traces in this case, but having one more combination of behaviors (assertions enabled but exception stack traces disabled) seems harder to manage and really has no point.

> How about just: `if settings.EXCEPTION_STACK_TRACES and settings.WASM_EXCEPTIONS:`?

I think this works fine. Now the code is:
```python
  if settings.ASSERTIONS:                                                        
    settings.EXCEPTION_STACK_TRACES = 1                                          
  if settings.EXCEPTION_STACK_TRACES and settings.WASM_EXCEPTIONS:   
    ...
```
Also I was able to change the condition in a few JS libraries to simply
```js
#if EXCEPTION_STACK_TRACES
```

> Or are the two lines below also needed when `ASSERTIONS` are enabled on their own? (Why?)

Not sure which two lines below you are talking about?